src/paramTypes.ts

```diff
@@ -18,8 +18,18 @@ export type DataObjectType = {
 	};
 }
 
+export type queueObjectType = {
+	name: string,
+	middleware?: middlewareFunc, 
+};
+
+export type middlewareFunc = 
+	(dataObject: DataObjectType, collection: string) => 
+	(null| string | Object | Buffer |string[] | Object[] 
+		| Buffer[] | undefined);
+
 export type MTROptions = {
 	silent: boolean;
 	prettify: boolean;
-	middleware: (datqaObject: DataObjectType) => (string | Object | Buffer);
-}
+	retries?: number;
```

Comment:
if the retries are about the connection to rabbit, perhaps call them rabbitRetries?

Reply:
done